CMakeLists.txt

```diff
@@ -207,15 +216,16 @@ if (ONEDPL_BACKEND MATCHES "^(tbb|dpcpp|dpcpp_only)$")
             message(STATUS "Using a default device backend (*)")
         endif()
 
+        if (INTEL_LLVM_COMPILER AND INTEL_LLVM_COMPILER_VERSION VERSION_LESS 2023.1)
+            set(DEVICE_SELECTION_VARIABLE "SYCL_DEVICE_FILTER")
+            set(FPGA_DEVICE_TYPE "acc")
+        else()
+            set(DEVICE_SELECTION_VARIABLE "ONEAPI_DEVICE_SELECTOR")
+            set(FPGA_DEVICE_TYPE "fpga")
+        endif()
+
         # set up device selection line basing on ONEDPL_DEVICE_BACKEND and ONEDPL_DEVICE_TYPE values
         if (DEFINED ONEDPL_DEVICE_BACKEND AND DEFINED ONEDPL_DEVICE_TYPE)
```

Comment:
Following up on above, will this code conflict with PSTL offload device selection?  (and should we prevent this in some way?)

Reply:
Thank you, that was a bug in `DEVICE_SELECTION_VARIABLE` usage. I added a check on `onedpl_add_test()` level.

Reply:
Looks good.